src/driving/vehicle_boat.py

```diff
@@ -0,0 +1,276 @@
+from __future__ import annotations
+
+import math
+
+import arcade
+
+import constants
+from driving.drive_mode import DriveMode
+
+DEFAULT_WORLD_SIZE_X = constants.SCREEN_WIDTH
+DEFAULT_WORLD_SIZE_Y = constants.SCREEN_HEIGHT
+DEFAULT_ACCELERATION_RATE = 2.5
+DEFAULT_DECELERATION_RATE = 0.1
+DEFAULT_BRAKE_RATE = 25
+DEFAULT_FRICTION = 25
+DEFAULT_DRIVE_SPEED = 12
+DEFAULT_TURN_SPEED = 2.2
+
+DEFAULT_CONTROL_LOCKOUT = 0.4
+
+IMPACT_FORCE_FLOOR = 1.5
+
+AXIS_FLOOR = 0.3
+
+
+class Boat(DriveMode):
```

Comment:
lol, what if it's an airship, so it can float over walls, but is hard to control?